smithy-model/src/main/java/software/amazon/smithy/model/loader/IdlTokenizer.java

```diff
@@ -0,0 +1,824 @@
+            int lineStart = parser.position();
+            parser.consumeRemainingCharactersOnLine();
+            docCommentLines.add(getInput(lineStart, parser.position()));
+        } else {
+            parser.consumeRemainingCharactersOnLine();
+        }
+
+        // Include the newline in the comment and doc comment lexeme.
+        if (parser.expect('\r', '\n', SimpleParser.EOF) == '\r' && parser.peek() == '\n') {
+            parser.skip();
+        }
+
+        currentTokenEnd = parser.position();
+        return currentTokenType = type;
+    }
+
+    /**
+     * Removes any buffered documentation comment lines, and returns a concatenated string value.
+     *
+     * @return Returns the combined documentation comment string for the given lines. Returns null if no lines.
+     */
+    public String removePendingDocCommentLines() {
+        if (docCommentLines.isEmpty()) {
+            return null;
+        } else {
+            StringBuilder result = new StringBuilder();
+            result.append(docCommentLines.removeFirst());
+            while (!docCommentLines.isEmpty()) {
+                result.append('\n').append(docCommentLines.removeFirst());
+            }
+            return result.toString();
+        }
+    }
+
+    private IdlToken parseNumber() {
+        try {
+            String lexeme = ParserUtils.parseNumber(parser);
+            if (lexeme.contains("e") || lexeme.contains("E")  || lexeme.contains(".")) {
+                double value = Double.parseDouble(lexeme);
+                if (Double.isFinite(value)) {
+                    currentTokenNumber = value;
+                } else {
+                    currentTokenNumber = new BigDecimal(lexeme);
+                }
+            } else {
+                try {
+                    currentTokenNumber = Long.parseLong(lexeme);
+                } catch (NumberFormatException e) {
+                    currentTokenNumber = new BigInteger(lexeme);
+                }
+            }
+
+            currentTokenEnd = parser.position();
+            return currentTokenType = IdlToken.NUMBER;
+        } catch (RuntimeException e) {
+            currentTokenEnd = parser.position();
+            // Strip off the leading error message information if present.
+            if (e.getMessage().startsWith("Syntax error")) {
+                currentTokenError = e.getMessage().substring(e.getMessage().indexOf(':') + 1).trim();
+            } else {
+                currentTokenError = e.getMessage();
+            }
+            return currentTokenType = IdlToken.ERROR;
+        }
+    }
+
+    private IdlToken parseIdentifier() {
+        ParserUtils.consumeIdentifier(parser);
+        currentTokenEnd = parser.position();
+        return currentTokenType = IdlToken.IDENTIFIER;
+    }
+
+    private IdlToken parseString() {
+        parser.skip();
+
+        if (parser.peek() == '"') {
+            parser.skip();
+            if (parser.peek() == '"') {
+                parser.skip();
```

Comment:
It won't, but I'll add a test for this. On line 759, we skip the first quote. Then on line 761, we skip the second.

Reply:
I got confused also but then I realized that the first quote is already skipped at the beginning of this methods, so this will match the second and third and the empty string will fall into the else branch.

Reply:
I added some comments